packages/shared/src/hooks/useChat.ts

```diff
@@ -0,0 +1,353 @@
+
+However, the true surprise came from Rusty. With an unexpected burst of energy, he overtook the others and crossed the finish line first. The crowd erupted in cheers, celebrating his triumphant comeback. Rusty couldn't believe it—his dream of being appreciated had come true.
+
+In the end, the Great Rally taught everyone a valuable lesson: it's not just about speed and victory, but about the love, companionship, and determination that made the Autozens special. From that day on, Rusty was no longer overlooked; he became a respected and cherished member of the Automatopia community.
+
+~~~javascript
+const a = 1;
+
+a += 1
+
+for (let i=0; i += 1; i < 10) {
+  console.log(a + i)
+}
+~~~
+
+#### Some rust
+
+~~~rust
+fn main() {
+  // Statements here are executed when the compiled binary is called.
+
+  // Print text to the console.
+  println!("Hello World!");
+}
+~~~
+
+## The end
+
+And so, in the city of Automatopia, the cars continued to live harmoniously with their human companions, cherishing the bonds they shared and forever embracing the spirit of adventure and unity.`;
+
+const queue = [
+  {
+    type: 'Type of the new event',
+    status: 'Current status of the process (optional)',
+    timestamp: 'Generation time of the event',
+    payload: 'Generic container extact object depends on the event type',
+  },
+  {
+    type: 'session_created',
+    status: 'Generating search query...',
+    payload: {
+      id: 'new session id (can be used to update the page url)',
+      steps: 'how many steps the process has',
+    },
+  },
+  {
+    type: 'web_search_finished',
+    status: 'Narrowing down search results...',
+    payload: {
+      sources: [],
+    },
+  },
+  {
+    type: 'web_results_filtered',
+    status: 'Scraping web pages...',
+    payload: {
+      ids: [],
+    },
+  },
+];
+
+const createStream = (content, chunkSize = 50) => {
+  let cursor = 0;
+  const stream = [];
+
+  while (cursor <= content.length) {
+    const chunk = content.slice(cursor, cursor + chunkSize);
+    stream.push({
+      type: 'new_token_received',
+      payload: {
+        token: chunk,
+      },
+    });
+    cursor += chunkSize;
+  }
+
+  return stream;
+};
+
+const mockChatStream = ({
+  onMessage,
+}: {
+  onMessage: (event: { data: string }) => void;
+}): (() => void) => {
+  let mounted = true;
+
+  const mockStream = async () => {
+    const contentStream = createStream(carStoryMarkdownTest).reverse();
+    const myQueue = [...queue].reverse();
+    const stream = [
+      {
+        type: 'completed',
+      },
+      ...contentStream,
+      ...myQueue,
+    ];
+
+    while (stream.length) {
+      // eslint-disable-next-line no-await-in-loop
+      await new Promise((resolve) => setTimeout(resolve, 400));
+
+      if (!mounted) {
+        break;
+      }
+
+      onMessage({
+        data: JSON.stringify(stream.pop()),
+      });
+    }
+  };
+
+  mockStream();
```

Comment:
We just need to ensure we maybe add a ticket for cleanup once we have implementation details

Reply:
It should be the `EventSource` implementation as mentioned in the DR so we can add it to that task.

Reply:
Linked in this task https://dailydotdev.atlassian.net/browse/WT-1626